### PR TITLE
[RLlib] Make sure unsquash_action moves user action to proper range

### LINKED ISSUE #23074: [Bug] Actions from outside action_space chosen
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

RLlib

### Issue Severity

High: It blocks me to complete my task.

### What happened + What you expected to happen

This seems to be a regression on bug https://github.com/ray-project/ray/issues/2965

When given a box action space of -1 to +1, the agent is given actions outside of the [low, high] range.

Notably, config['clip_actions'] = True which as I understand it is the fix to the previously referenced issue.

### Versions / Dependencies

ray==1.11.0
ray-cpp==1.11.0 (from pypi)

Python 3.9.10 (cpython from homebrew)

MacOS Monterey 12.1 (x86_64, not arm)

### Reproduction script
```
import gym
import numpy as np
class BuggedEnv(gym.Env):
    def __init__(self, config):
        self.action_space = gym.spaces.Box(low=-1, high=1, shape=(100,), dtype=np.int32)
        self.observation_space = gym.spaces.Box(low=-10, high=10, shape=(40,), dtype=np.int32)
    def step(self, action):
        for idx, value in enumerate(action):
            assert -1 <= value <= 1, f"action {idx=} {value=} outside action space"
        return np.zeros(self.observation_space.shape, dtype=np.int32), 0, True, {}
    def reset(self):
        return np.zeros(self.observation_space.shape, dtype=np.int32)
import ray
ray.init()
import ray.rllib.agents.ppo as ppo
config = ppo.DEFAULT_CONFIG.copy()
config['clip_actions'] = True
agent = ppo.PPOTrainer(env=BuggedEnv, config=config)
agent.train() # Triggers assert on line 9
```
### Anything else

_No response_

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by gjoliver**
## Why are these changes needed?

for int Box space as well.

## Related issue number

Closes #23074

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: rllib/utils/spaces/space_utils.py [hunk 1]
```diff
@@ -256,18 +256,18 @@ def unsquash_action(action, action_space_struct):
...
-            a = np.clip(a, s.low, s.high)
+        if isinstance(s, gym.spaces.Box) and np.all(s.bounded_below) and np.all(s.bounded_above):
+            if s.dtype == np.float32 or s.dtype == np.float64:
+                # Assuming values are roughly between -1.0 and 1.0 ->
+                # unsquash them to the given bounds.
+                a = s.low + (a + 1.0) * (s.high - s.low) / 2.0
+                # Clip to given bounds, just in case the squashed values were
+                # outside [-1.0, 1.0].
+                a = np.clip(a, s.low, s.high)
+            elif s.dtype == np.int32 or s.dtype == np.int64:
```
Comment (ArturNiederfahrenhorst) [inline rllib/utils/spaces/space_utils.py:267]: The comments make reading this super easy :)
Should we check for np.integer here instead?
np.issubdtype(s.dtype, np.integer)?

**Review by ArturNiederfahrenhorst — COMMENTED** [review]
Great!

File: rllib/utils/spaces/space_utils.py (hunk 1 above)
Comment (gjoliver) [reply rllib/utils/spaces/space_utils.py]: done. it's weird, why doesn't
```
np.issubdtype(s.dtype, np.float)
```
work for float32 and float64??

Comment (ArturNiederfahrenhorst) [reply rllib/utils/spaces/space_utils.py]: Huh. Works on my machine.

Comment (ArturNiederfahrenhorst) [reply rllib/utils/spaces/space_utils.py]: s = np.array([0, 1], dtype=np.float64)
np.issubdtype(s.dtype, np.float)
True

Comment (ArturNiederfahrenhorst) [reply rllib/utils/spaces/space_utils.py]: numpy==1.22.3

Comment (gjoliver) [reply rllib/utils/spaces/space_utils.py]: > numpy==1.22.3

right, but do you know why float32 doesn't work?
```
>>> s = np.array([0, 1], dtype=np.float32)
>>> np.issubdtype(s.dtype, np.float)
False
```

**Review by sven1977 — APPROVED** [review]
Looks great! Thanks for the fix @gjoliver and your review @ArturNiederfahrenhorst !

**Pull request merged**